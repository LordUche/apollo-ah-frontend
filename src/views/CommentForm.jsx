
import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * Creates the JSX element used to display a form with input to post a new comment
 * @param {object} props The component props
 * handleInputChange -  The function to bind the input field to the corresponding state value
 * handlePostComment - The function to execute to post a comment
 * postingComment - A boolean keeping track of the status of the post comment action
 * body - The value of the input field
 * isLoggedIn - Used to force unauthenticated users from entering comments
 * @returns {Node} The generated JSX element
 */
export default function PostComment({
  handlePostComment,
  postingComment,
  handleInputChange,
  body,
  isLoggedIn,
}) {
  return (
    <div className="comment-grp card mt-3 relative">
      <form className={`form${postingComment ? ' busy' : ''}`} onSubmit={handlePostComment}>
        <div className="form-group">
          <input className="form-control" autoComplete="off" disabled={postingComment} placeholder="Leave a comment" name="body" onChange={handleInputChange} value={body} />
          <button type="submit" className="btn btn-link" disabled={postingComment}>
            <i className="fas fa-paper-plane" />
            <span className="spinner-grow text-dark" />
          </button>
        </div>
      </form>
      {!isLoggedIn && <Link to="/login" className="link-surface" />}
    </div>
  );
}

/** PropTypes declarations */
PostComment.propTypes = {
  handlePostComment: PropTypes.func.isRequired,
  postingComment: PropTypes.bool.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  body: PropTypes.string.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
};
