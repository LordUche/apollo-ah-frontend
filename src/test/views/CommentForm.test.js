import React from 'react';
import { shallow } from 'enzyme';
import CommentForm from '../../views/CommentForm';

const mockFn = jest.fn();

describe('<CommentForm>', () => {
  it('should render without crashing when user is unauthenticated', () => {
    const wrapper = shallow(<CommentForm
      postComment={mockFn}
      handlePostComment={mockFn}
      postingComment={false}
      handleInputChange={mockFn}
      body="Hello world"
      isLoggedIn={false}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when user is authenticated', () => {
    const wrapper = shallow(<CommentForm
      postComment={mockFn}
      handlePostComment={mockFn}
      postingComment
      handleInputChange={mockFn}
      body="Hello world"
      isLoggedIn
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
