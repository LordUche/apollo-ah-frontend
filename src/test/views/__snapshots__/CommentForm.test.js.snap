// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm> should render without crashing when user is authenticated 1`] = `
<div
  className="comment-grp card mt-3 relative"
>
  <form
    className="form busy"
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        disabled={true}
        name="body"
        onChange={[MockFunction]}
        placeholder="Leave a comment"
        value="Hello world"
      />
      <button
        className="btn btn-link"
        disabled={true}
        type="submit"
      >
        <i
          className="fas fa-paper-plane"
        />
        <span
          className="spinner-grow text-dark"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`<CommentForm> should render without crashing when user is unauthenticated 1`] = `
<div
  className="comment-grp card mt-3 relative"
>
  <form
    className="form"
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        disabled={false}
        name="body"
        onChange={[MockFunction]}
        placeholder="Leave a comment"
        value="Hello world"
      />
      <button
        className="btn btn-link"
        disabled={false}
        type="submit"
      >
        <i
          className="fas fa-paper-plane"
        />
        <span
          className="spinner-grow text-dark"
        />
      </button>
    </div>
  </form>
  <Link
    className="link-surface"
    replace={false}
    to="/login"
  />
</div>
`;
